azure-core/azure/core/paging.py

```diff
@@ -0,0 +1,113 @@
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the ""Software""), to
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+import sys
+try:
+    from collections.abc import Iterator
+    xrange = range
+except ImportError:
+    from collections import Iterator
+
+from typing import Dict, Any, List, Callable, Optional, TYPE_CHECKING  # pylint: disable=unused-import
+
+from msrest.serialization import Deserializer
+
+if TYPE_CHECKING:
+    from .pipeline.transport import HttpResponse  # pylint: disable=unused-import
+    from msrest.serialization import Model  # pylint: disable=unused-import
+
+if sys.version_info >= (3, 5, 2):
+    # Not executed on old Python, no syntax error
+    from .async_paging import AsyncPagedMixin  # type: ignore
+else:
+    class AsyncPagedMixin(object):  # type: ignore
+        pass
+
+class Paged(AsyncPagedMixin, Iterator):
+    """A container for paged REST responses.
+
+    :param HttpResponse response: server response object.
+    :param callable command: Function to retrieve the next page of items.
+    :param dict classes: A dictionary of class dependencies for
+     deserialization.
+    :param dict raw_headers: A dict of raw headers to add if "raw" is called
+    """
+    _validation = {}  # type: Dict[str, Dict[str, Any]]
+    _attribute_map = {}  # type: Dict[str, Dict[str, Any]]
+
+    def __init__(self, command, classes, raw_headers=None, **kwargs):
+        # type: (Callable[[str], HttpResponse], Dict[str, Model], Dict[str, str], Any) -> None
+        super(Paged, self).__init__(**kwargs)  # type: ignore
+        # Sets next_link, current_page, and _current_page_iter_index.
+        self.next_link = ""
+        self.current_page = []  # type: List[Model]
+        self._current_page_iter_index = 0
+        self._deserializer = Deserializer(classes)
+        self._get_next = command
+        self._response = None  # type: Optional[HttpResponse]
+        self._raw_headers = raw_headers
```

Comment:
Not sure we need this any more. Though I guess well know for sure once we get the autorest tests running.

Reply:
addressed in https://github.com/Azure/azure-sdk-for-python/pull/4873/commits/80ae908ada26ef585a7437a3b6e5c58461a9e0ad (note I also removed the initializer parameter)